core/unit_test/TestPrintf.hpp

```diff
@@ -0,0 +1,88 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <gtest/gtest.h>
+
+#include <Kokkos_Core.hpp>
+#include <fstream>
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+template <class ExecutionSpace>
+void test_kokkos_printf() {
+#ifdef _WIN32
+  char* tmp_filename = tmpnam(nullptr);
+  ASSRET_NE(tmp_filename, nullptr);
+#else
+  char tmp_filename[16] = "/tmp/XXXXXXXXXX";
+  int error             = mkstemp(tmp_filename);
+  ASSERT_NE(error, -1);
+#endif
+
+  int backup_no, new_no;
+  fflush(stdout);
```

Comment:
Any particular reason why you are not using `::testing::internal::CaptureStdout()`
It is not part of the GTest API but we already use it elsewhere and if/when we get into trouble for it we can encapsulate and replace.

Reply:
I didn't quite realize that we are already reaching into someone else's `internal` namespace. Sure, we can try that instead. It's just for testing anyway.